lib/Mojolicious/Controller.pm

```diff
@@ -0,0 +1,886 @@
+
+  # Make sure response is cached correctly
+  $c->res->headers->cache_control('public, max-age=300');
+  $c->res->headers->append(Vary => 'Accept-Encoding');
+
+=head2 send
+
+  $c = $c->send({binary => $bytes});
+  $c = $c->send({text   => $bytes});
+  $c = $c->send({json   => {test => [1, 2, 3]}});
+  $c = $c->send([$fin, $rsv1, $rsv2, $rsv3, $op, $payload]);
+  $c = $c->send($chars);
+  $c = $c->send($chars => sub ($c) {...});
+
+Send message or frame non-blocking via WebSocket, the optional drain callback will be executed once all data has been
+written. This method will automatically respond to WebSocket handshake requests with a C<101> response status, to
+establish the WebSocket connection.
+
+  # Send "Text" message
+  $c->send('I ♥ Mojolicious!');
+
+  # Send JSON object as "Text" message
+  $c->send({json => {test => 'I ♥ Mojolicious!'}});
+
+  # Send JSON object as "Binary" message
+  use Mojo::JSON qw(encode_json);
+  $c->send({binary => encode_json({test => 'I ♥ Mojolicious!'})});
+
+  # Send "Ping" frame
+  use Mojo::WebSocket qw(WS_PING);
+  $c->send([1, 0, 0, 0, WS_PING, 'Hello World!']);
+
+  # Make sure the first message has been written before continuing
+  $c->send('First message!' => sub ($c) { $c->send('Second message!') });
+
+For mostly idle WebSockets you might also want to increase the inactivity timeout with
+L<Mojolicious::Plugin::DefaultHelpers/"inactivity_timeout">, which usually defaults to C<30> seconds.
+
+  # Increase inactivity timeout for connection to 300 seconds
+  $c->inactivity_timeout(300);
+
+=head2 session
+
+  my $session = $c->session;
+  my $foo     = $c->session('foo');
+  $c          = $c->session({foo => 'bar'});
+  $c          = $c->session(foo => 'bar');
+
+Persistent data storage for the next few requests, all session data gets serialized with L<Mojo::JSON> and stored
+Base64 encoded in HMAC-SHA256 signed cookies, to prevent tampering. Note that cookies usually have a C<4096> byte
+(4KiB) limit, depending on browser.
+
+  # Manipulate session
+  $c->session->{foo} = 'bar';
+  my $foo = $c->session->{foo};
+  delete $c->session->{foo};
+
+  # Expiration date in seconds from now (persists between requests)
+  $c->session(expiration => 604800);
+
+  # Expiration date as absolute epoch time (only valid for one request)
+  $c->session(expires => time + 604800);
+
+  # Delete whole session by setting an expiration date in the past
+  $c->session(expires => 1);
+
+=head2 signed_cookie
+
+  my $value = $c->signed_cookie('foo');
+  $c        = $c->signed_cookie(foo => 'bar');
+  $c        = $c->signed_cookie(foo => 'bar', {path => '/'});
+
+Access signed request cookie values and create new signed response cookies. If there are multiple values sharing the
+same name, and you want to access more than just the last one, you can use L</"every_signed_cookie">. Cookies are
+cryptographically signed with HMAC-SHA256, to prevent tampering, and the ones failing signature verification will be
```

Comment:
This is really great!  Separately, wouldn't it be nice for Mojo to also offer an *encrypted* session?  Is there a reason not to?  Is it because there is no Perl core encryption?  What about made available if there is an optional module installed?  Perhaps the issue then is what if the optional module isn't available... what happens?  Would the app fail to run?  Would the app fallback to just signed (that's not acceptable obviously).  Is there simply no way to offer an `encyrypted_cookie` function?